external/concealer/src/autoware_universe.cpp

```diff
@@ -0,0 +1,267 @@
+    setGoalPose(route.back());
+    for (const auto & each : route | boost::adaptors::sliced(0, route.size() - 1)) {
+      setCheckpoint(each);
+    }
+    waitForAutowareStateToBePlanning();
+    waitForAutowareStateToBeWaitingForEngage();  // NOTE: Autoware.IV 0.11.1 waits about 3 sec from the completion of Planning until the transition to WaitingForEngage.
+  });
+}
+
+auto AutowareUniverse::engage() -> void
+{
+  task_queue.delay([this]() {
+    waitForAutowareStateToBeDriving([this]() {
+      AutowareEngage message;
+      {
+        message.stamp = get_clock()->now();
+        message.engage = true;
+      }
+      return setAutowareEngage(message);
+    });
+  });
+}
+
+auto AutowareUniverse::update() -> void
+{
+  CurrentControlMode current_control_mode;
+  {
+    current_control_mode.mode = CurrentControlMode::AUTONOMOUS;
+  }
+  setCurrentControlMode(current_control_mode);
+
+  CurrentShift current_shift;
+  {
+    using autoware_auto_vehicle_msgs::msg::GearReport;
+    current_shift.stamp = get_clock()->now();
+    current_shift.report = current_twist.linear.x >= 0 ? GearReport::DRIVE : GearReport::REVERSE;
+  }
+  setCurrentShift(current_shift);
+
+  CurrentSteering current_steering;
+  {
+    current_steering.stamp = get_clock()->now();
+    current_steering.steering_tire_angle = getSteeringAngle();
+  }
+  setCurrentSteering(current_steering);
+
+  CurrentVelocity current_velocity;
+  {
+    current_velocity.header.stamp = get_clock()->now();
+    current_velocity.header.frame_id = "base_link";
+    current_velocity.longitudinal_velocity = current_twist.linear.x;
+    current_velocity.lateral_velocity = current_twist.linear.y;
+    current_velocity.heading_rate = current_twist.angular.z;
+  }
+  setCurrentVelocity(current_velocity);
+
+  LocalizationOdometry localization_odometry;
+  {
+    localization_odometry.header.stamp = get_clock()->now();
+    localization_odometry.header.frame_id = "map";
+    localization_odometry.pose.pose = current_pose;
+    localization_odometry.pose.covariance = {};
+    localization_odometry.twist.twist = current_twist;
+  }
+  setLocalizationOdometry(localization_odometry);
+
+  VehicleVelocity vehicle_velocity;
+  {
+    vehicle_velocity.stamp = get_clock()->now();
+    vehicle_velocity.max_velocity = current_upper_bound_speed;
+  }
+  setVehicleVelocity(vehicle_velocity);
+
+  setTransform(current_pose);
+}
+
+auto AutowareUniverse::getAcceleration() const -> double
+{
+  return getAckermannControlCommand().longitudinal.acceleration;
+}
+
+auto AutowareUniverse::getVelocity() const -> double
+{
+  return getAckermannControlCommand().longitudinal.speed;
+}
+
+auto AutowareUniverse::getSteeringAngle() const -> double
+{
+  return getAckermannControlCommand().lateral.steering_tire_angle;
+}
+
+auto AutowareUniverse::getGearSign() const -> double
+{
+  using autoware_auto_vehicle_msgs::msg::GearCommand;
+  return getGearCommand().command == GearCommand::REVERSE ? -1.0 : 1.0;
```

Comment:
```
getGearCommand().command == GearCommand::REVERSE
```

This is not an assignment.

Reply:
Sorry for my mistake, but what is gear sign?
Is it gear position of something?
Is it a continuous value?

Reply:
In order to tell the vehicle model whether it is reverse gear or not (= forward or backward), we need to get a positive or negative sign instead of an enum. The numerical value (= +1.0 or -1.0) obtained from getGearSign is used as the control input of the vehicle model and is used for multiplication.